File: backend/models/youtube.py
from pydantic import BaseModel
from typing import Optional

class UrlData(BaseModel):
    url: str
    downloadType: str | None = None 

class YoutubeInfo(BaseModel):
    title: str
    description: str
    duration: Optional[str]
    thumbnail: str
    videoId : Optional[str]

File: dwnldr/app/types/youtube.ts
export type YoutubeResult = {
  thumbnail: string;
  title: string;
  description: string;
  duration: string;
  videoId?: string;
};

export type DownloadType = "video" | "audio";

export interface TrimSliderProps {
  key?: string;
  videoId?: string;
  duration: number | string;
  url: string;
  downloadType: "video" | "audio";
}
